File: debug_db.py
import os
import psycopg2
from dotenv import load_dotenv

# Load env from the api directory
load_dotenv('/Users/tengjunsiong/Projects/ktmb0t/Stack/ktmb_timetable/ktmb_api/.env')

def check_db():
    try:
        db_url = os.getenv('DATABASE_URL')
        if not db_url:
            print("DATABASE_URL not found in .env")
            return

        conn = psycopg2.connect(db_url)
        cursor = conn.cursor()

        print("--- Routes ---")
        cursor.execute("SELECT id, name, service_type FROM routes WHERE service_type = 'ETS'")
        routes = cursor.fetchall()
        print(f"Found {len(routes)} ETS routes:")
        for r in routes:
            print(r)
            
        if not routes:
            print("No ETS routes found. Checking all service types...")
            cursor.execute("SELECT DISTINCT service_type FROM routes")
            print(cursor.fetchall())
            return

        print("\n--- Schedules for 28 Dec 2025 ---")
        # Check if we have valid schedules covering this date
        cursor.execute("""
            SELECT s.id, s.valid_from, s.valid_to, s.day_type, r.name 
            FROM schedules s 
            JOIN routes r ON s.route_id = r.id 
            WHERE r.service_type = 'ETS'
        """)
        schedules = cursor.fetchall()
        print(f"Found {len(schedules)} ETS schedules:")
        for s in schedules:
            print(s)

        conn.close()

    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    check_db()

File: debug_db.js
require('dotenv').config({ path: '/Users/tengjunsiong/Projects/ktmb0t/Stack/ktmb_timetable/ktmb_api/.env' });
const db = require('./ktmb_api/src/config/db');

const checkDB = async () => {
    try {
        console.log("Checking DB...");

        console.log("\n--- Service Types ---");
        const res1 = await db.query("SELECT DISTINCT service_type FROM routes");
        console.log(res1.rows);

        console.log("\n--- ETS Routes ---");
        const res2 = await db.query("SELECT id, name, service_type FROM routes WHERE service_type = 'ETS'");
        console.log(res2.rows);

        if (res2.rows.length > 0) {
            console.log("\n--- Schedules for ETS ---");
            const res3 = await db.query(`
                SELECT s.id, s.valid_from, s.valid_to, s.day_type, r.name 
                FROM schedules s 
                JOIN routes r ON s.route_id = r.id 
                WHERE r.service_type = 'ETS'
            `);
            console.log(res3.rows);
        }

    } catch (err) {
        console.error("Error:", err);
    } finally {
        // We can't easily close the pool if not exposed, but script will end.
        // Actually db module is usually a Pool object.
        process.exit();
    }
};

checkDB();
